src/tests/posix/pthread/pthread_kill_test.c

```diff
@@ -0,0 +1,121 @@
+/**
+ * @file
+ * @brief test for pthread_kill
+ *
+ * @author  XU Chenglin
+ * @date    09/03/2020
+ */
+
+#include <pthread.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+#include <signal.h>
+#include <embox/test.h>
+
+
+EMBOX_TEST_SUITE("posix/pthread_kill");
+
+
+void sighand(int signo)
+{
+    return;
+}
+
+void *thread_test(void *parm)
+{
+    int rc = sleep(10);
+    test_assert_not_zero(rc);
+    return NULL;
+}
+
+void *thread_ref(void *parm)
+{
+    int rc = sleep(10);
+    test_assert_zero(rc);
+    return NULL;
+}
+
+void *thread_immediate_return(void *parm)
+{
+    return NULL;
+}
+
+TEST_CASE("Send signal to a running thread") {
+    int rc;
+    struct sigaction actions;
+    pthread_t thread_normal, thread_blind;
+
+    sigemptyset(&actions.sa_mask);
+    actions.sa_flags = 0;
+    actions.sa_handler = sighand;
+
+    rc = sigaction(SIGALRM,&actions,NULL);
+
+    rc = pthread_create(&thread_normal, NULL, thread_test, NULL);
+    if (rc != 0) {
+        test_fail("Failed pthread_create\n");
+    }
+
+    rc = pthread_create(&thread_blind, NULL, thread_ref, NULL);
+    if (rc != 0) {
+        test_fail("Failed pthread_create\n");
+    }
+
+    rc = pthread_kill(thread_normal, SIGALRM);
+    test_assert_zero(rc);
+    rc = pthread_join(thread_normal, NULL);
+    rc = pthread_join(thread_blind, NULL);
+}
+
+/*
+* Currently, using pthread_kill to send signal to terminated threads
+* behaves wierd. If the signal sent is SIGALRM, the program will crash.
```

Comment:
Can you investigate it with more details?

Reply:
I will dig deeper and send you a mail when I finish.

Reply:
Ok